[test_code_review] Failing with `StopIteration` error when running on Taskcluster

The test passes when run locally.

https://community-tc.services.mozilla.com/tasks/StRnqd9JQA6onTN_21OcGA/runs/0/logs/public/logs/live.log:
```
updating to branch default
1289 files updated, 0 files merged, 0 files removed, 0 files unresolved
+ python -m pytest --cov=./ tests/test_assignee.py tests/test_backout.py tests/test_bug.py tests/test_bug_features.py tests/test_bug_snapshot.py tests/test_bugtype.py tests/test_bugzilla.py tests/test_code_review.py tests/test_commit_features.py tests/test_db.py tests/test_defect.py tests/test_defect_enhancement_task.py tests/test_devdocneeded.py tests/test_feature_cleanup.py tests/test_github.py tests/test_github_issue_retriever.py tests/test_hooks.py tests/test_invalid_compatibility_report.py tests/test_labels.py tests/test_models.py tests/test_needsdiagnosis.py tests/test_performancebug.py tests/test_phabricator.py tests/test_pipelines.py tests/test_qaneeded.py tests/test_rcatype.py tests/test_regression.py tests/test_repository.py tests/test_stepstoreproduce.py tests/test_test_scheduling.py tests/test_test_scheduling_features.py tests/test_testselect.py tests/test_tracking.py tests/test_trainer.py tests/test_uplift.py tests/test_utils.py
============================= test session starts ==============================
platform linux -- Python 3.12.3, pytest-8.4.1, pluggy-1.6.0
rootdir: /bugbug
configfile: pyproject.toml
plugins: hypothesis-6.138.6, responses-0.5.1, langsmith-0.4.26, cov-6.2.1, anyio-4.10.0
collecting ... 
collecting 1 item                                                              
collecting 32 items                                                            
collecting 134 items                                                           
collecting 160 items                                                           
collected 183 items                                                            
tests/test_assignee.py .                                                 [  0%]
tests/test_backout.py .                                                  [  1%]
tests/test_bug.py .                                                      [  1%]
tests/test_bug_features.py .......................                       [ 14%]
tests/test_bug_snapshot.py .                                             [ 14%]
tests/test_bugtype.py .                                                  [ 15%]
tests/test_bugzilla.py ...                                               [ 16%]
tests/test_code_review.py F                                              [ 17%]
tests/test_commit_features.py .                                          [ 18%]
tests/test_db.py ...................................                     [ 37%]
tests/test_defect.py .                                                   [ 37%]
tests/test_defect_enhancement_task.py .                                  [ 38%]
tests/test_devdocneeded.py .                                             [ 38%]
tests/test_feature_cleanup.py ........                                   [ 43%]
tests/test_github.py ........                                            [ 47%]
tests/test_github_issue_retriever.py ....                                [ 49%]
tests/test_hooks.py ......                                               [ 53%]
tests/test_invalid_compatibility_report.py .                             [ 53%]
tests/test_labels.py ..                                                  [ 54%]
tests/test_models.py ...                                                 [ 56%]
tests/test_needsdiagnosis.py .                                           [ 56%]
tests/test_performancebug.py .                                           [ 57%]
tests/test_phabricator.py .                                              [ 57%]
tests/test_pipelines.py ...                                              [ 59%]
tests/test_qaneeded.py ..                                                [ 60%]
tests/test_rcatype.py ..                                                 [ 61%]
tests/test_regression.py .                                               [ 62%]
tests/test_repository.py ....................                            [ 73%]
tests/test_stepstoreproduce.py .                                         [ 73%]
tests/test_test_scheduling.py ........                                   [ 78%]
tests/test_test_scheduling_features.py .                                 [ 78%]
tests/test_testselect.py ..............x.                                [ 87%]
tests/test_tracking.py .                                                 [ 87%]
tests/test_trainer.py .                                                  [ 88%]
tests/test_uplift.py .                                                   [ 89%]
tests/test_utils.py ....................                                 [100%]
=================================== FAILURES ===================================
___________________________ test_find_comment_scope ____________________________
    def test_find_comment_scope():
        responses.add_passthru("https://phabricator.services.mozilla.com/")
        responses.add_passthru(
            "https://mozphab-phabhost-cdn.devsvcprod.mozaws.net/file/data/"
        )
    
        test_data = {
            "https://phabricator.services.mozilla.com/D233024?id=964198": {
                "browser/components/newtab/test/browser/browser.toml": {
                    79: {
                        "line_start": 78,
                        "line_end": 79,
                        "has_added_lines": False,
                    }
                },
                "browser/components/asrouter/tests/browser/browser.toml": {
                    63: {
                        "line_start": 60,
                        "line_end": 74,
                        "has_added_lines": True,
                    },
                },
            },
            "https://phabricator.services.mozilla.com/D240754?id=995999": {
                "dom/canvas/WebGLShaderValidator.cpp": {
                    39: {
                        "line_start": 37,
                        "line_end": 42,
                        "has_added_lines": True,
                    },
                    46: {
                        "line_start": 37,
                        "line_end": 42,
                        "has_added_lines": True,
                    },
                }
            },
        }
    
        for revision_url, patch_files in test_data.items():
            raw_diff = requests.get(revision_url + "&download=true", timeout=5).text
            patch_set = PatchSet.from_string(raw_diff)
    
            for file_name, target_hunks in patch_files.items():
>               patched_file = next(
                    patched_file
                    for patched_file in patch_set
                    if patched_file.path == file_name
                )
E               StopIteration
/bugbug/tests/test_code_review.py:52: StopIteration
```

--- a/tests/fixtures/phabricator/D233024-964198.diff
+++ b/tests/fixtures/phabricator/D233024-964198.diff
@@ -1,0 +1,95 @@
+diff --git a/browser/components/asrouter/tests/browser/browser.toml b/browser/components/asrouter/tests/browser/browser.toml
+--- a/browser/components/asrouter/tests/browser/browser.toml
++++ b/browser/components/asrouter/tests/browser/browser.toml
+@@ -55,11 +55,20 @@
+   "os == 'mac' && os_version == '14.70' && processor == 'x86_64' && debug", # Bug 1932912
+ ]
+
+ ["browser_feature_callout_panel.js"]
+
++["browser_foxdoodle_set_default.js"]
++
++["browser_multistage_spotlight.js"]
++
++["browser_multistage_spotlight_telemetry.js"]
++skip-if = ["verify"] # bug 1834620 - order of events not stable
++
+ ["browser_remote_l10n.js"]
+
+ ["browser_spotlight_default_messages.js"]
+
+ ["browser_trigger_listeners.js"]
+ https_first_disabled = true
++
++["browser_trigger_messagesLoaded.js"]
+diff --git a/browser/components/newtab/test/browser/browser_foxdoodle_set_default.js b/browser/components/asrouter/tests/browser/browser_foxdoodle_set_default.js
+rename from browser/components/newtab/test/browser/browser_foxdoodle_set_default.js
+rename to browser/components/asrouter/tests/browser/browser_foxdoodle_set_default.js
+diff --git a/browser/components/newtab/test/browser/browser_multistage_spotlight.js b/browser/components/asrouter/tests/browser/browser_multistage_spotlight.js
+rename from browser/components/newtab/test/browser/browser_multistage_spotlight.js
+rename to browser/components/asrouter/tests/browser/browser_multistage_spotlight.js
+diff --git a/browser/components/newtab/test/browser/browser_multistage_spotlight_telemetry.js b/browser/components/asrouter/tests/browser/browser_multistage_spotlight_telemetry.js
+rename from browser/components/newtab/test/browser/browser_multistage_spotlight_telemetry.js
+rename to browser/components/asrouter/tests/browser/browser_multistage_spotlight_telemetry.js
+diff --git a/browser/components/newtab/test/browser/browser_trigger_messagesLoaded.js b/browser/components/asrouter/tests/browser/browser_trigger_messagesLoaded.js
+rename from browser/components/newtab/test/browser/browser_trigger_messagesLoaded.js
+rename to browser/components/asrouter/tests/browser/browser_trigger_messagesLoaded.js
+diff --git a/browser/components/newtab/test/browser/browser.toml b/browser/components/newtab/test/browser/browser.toml
+--- a/browser/components/newtab/test/browser/browser.toml
++++ b/browser/components/newtab/test/browser/browser.toml
+@@ -37,21 +37,14 @@
+
+ ["browser_discovery_render.js"]
+
+ ["browser_enabled_newtabpage.js"]
+
+-["browser_foxdoodle_set_default.js"]
+-
+ ["browser_getScreenshots.js"]
+
+ ["browser_highlights_section.js"]
+
+-["browser_multistage_spotlight.js"]
+-
+-["browser_multistage_spotlight_telemetry.js"]
+-skip-if = ["verify"] # bug 1834620 - order of events not stable
+-
+ ["browser_newtab_glean.js"]
+
+ ["browser_newtab_header.js"]
+
+ ["browser_newtab_last_LinkMenu.js"]
+@@ -73,7 +66,5 @@
+ ["browser_topsites_annotation.js"]
+
+ ["browser_topsites_contextMenu_options.js"]
+
+ ["browser_topsites_section.js"]
+-
+-["browser_trigger_messagesLoaded.js"]
+diff --git a/browser/components/newtab/test/browser/head.js b/browser/components/newtab/test/browser/head.js
+--- a/browser/components/newtab/test/browser/head.js
++++ b/browser/components/newtab/test/browser/head.js
+@@ -1,21 +1,11 @@
+ "use strict";
+
+ ChromeUtils.defineESModuleGetters(this, {
+-  ASRouter: "resource:///modules/asrouter/ASRouter.sys.mjs",
+-
+   DiscoveryStreamFeed:
+     "resource://activity-stream/lib/DiscoveryStreamFeed.sys.mjs",
+
+-  FeatureCallout: "resource:///modules/asrouter/FeatureCallout.sys.mjs",
+-
+-  FeatureCalloutBroker:
+-    "resource:///modules/asrouter/FeatureCalloutBroker.sys.mjs",
+-
+-  FeatureCalloutMessages:
+-    "resource:///modules/asrouter/FeatureCalloutMessages.sys.mjs",
+-
+   ObjectUtils: "resource://gre/modules/ObjectUtils.sys.mjs",
+   PlacesTestUtils: "resource://testing-common/PlacesTestUtils.sys.mjs",
+   QueryCache: "resource:///modules/asrouter/ASRouterTargeting.sys.mjs",
+ });
+
+

--- a/tests/fixtures/phabricator/D240754-995999.diff
+++ b/tests/fixtures/phabricator/D240754-995999.diff
@@ -1,0 +1,28 @@
+diff --git a/dom/canvas/WebGLShaderValidator.cpp b/dom/canvas/WebGLShaderValidator.cpp
+--- a/dom/canvas/WebGLShaderValidator.cpp
++++ b/dom/canvas/WebGLShaderValidator.cpp
+@@ -33,19 +33,15 @@
+   options.enforcePackingRestrictions = true;
+   options.objectCode = true;
+   options.initGLPosition = true;
+   options.initializeUninitializedLocals = true;
+   options.initOutputVariables = true;
+-
+-#ifdef XP_MACOSX
+-  options.removeInvariantAndCentroidForESSL3 = true;
+-#else
+-  // We want to do this everywhere, but to do this on Mac, we need
+-  // to do it only on Mac OSX > 10.6 as this causes the shader
+-  // compiler in 10.6 to crash
+   options.clampIndirectArrayBounds = true;
+-#endif
++
++  if (kIsMacOS) {
++    options.removeInvariantAndCentroidForESSL3 = true;
++  }
+
+   if (gl->WorkAroundDriverBugs()) {
+     if (kIsMacOS) {
+       // Work around https://bugs.webkit.org/show_bug.cgi?id=124684,
+       // https://chromium.googlesource.com/angle/angle/+/5e70cf9d0b1bb
+

--- a/tests/test_code_review.py
+++ b/tests/test_code_review.py
@@ -1,18 +1,15 @@
-import requests
-import responses
+import os
+
 from unidiff import PatchSet
 
 from bugbug.tools.code_review import find_comment_scope
 
+FIXTURES_DIR = os.path.join(os.path.dirname(__file__), "fixtures/phabricator")
+
 
 def test_find_comment_scope():
-    responses.add_passthru("https://phabricator.services.mozilla.com/")
-    responses.add_passthru(
-        "https://mozphab-phabhost-cdn.devsvcprod.mozaws.net/file/data/"
-    )
-
     test_data = {
-        "https://phabricator.services.mozilla.com/D233024?id=964198": {
+        (233024, 964198): {
             "browser/components/newtab/test/browser/browser.toml": {
                 79: {
                     "line_start": 78,
@@ -28,7 +25,7 @@ def test_find_comment_scope():
                 },
             },
         },
-        "https://phabricator.services.mozilla.com/D240754?id=995999": {
+        (240754, 995999): {
             "dom/canvas/WebGLShaderValidator.cpp": {
                 39: {
                     "line_start": 37,
@@ -44,8 +41,10 @@ def test_find_comment_scope():
         },
     }
 
-    for revision_url, patch_files in test_data.items():
-        raw_diff = requests.get(revision_url + "&download=true", timeout=5).text
+    for (revision_id, diff_id), patch_files in test_data.items():
+        with open(os.path.join(FIXTURES_DIR, f"D{revision_id}-{diff_id}.diff")) as f:
+            raw_diff = f.read()
+
         patch_set = PatchSet.from_string(raw_diff)
 
         for file_name, target_hunks in patch_files.items():

